### PR TITLE
fix: toast ref update when showing a modal

### LINKED ISSUE #279: After toast shown within Modal, main instance outside of Modal doesn't work anymore
First of all, this lib is fantastic. And v2 is great. Thank you!

**Describe the bug**
I have two `Toast` instances setup. One alongside `NavigationContainer` from `react-navigation`, and one within `Modal`.

When I show a toast outside of the modal, it shows fine. And when I show a toast inside of the Modal when it's visible, it works fine. However, when I try to show a toast again outside of the Modal, it doesn't show. Bringing up the Modal again shows the modals within the modal fine still. 

So it's almost as if it's not tracking the correct instance and only tracks the Modal instance once Modal shows a toast.

**Steps to reproduce**
1. Show a toast when a Modal is not visible.
2. Make a Modal visible.
3. Show a toast. It shows on top of the Modal.
4. Make the modal not visible anymore.
5. Show a toast. It doesn't show.
6. Make the Modal visible again. It shows on top of the Modal.

**Expected behavior**
The toasts once a modal is no longer visible should show/hide properly.

**Screenshots**
This video shows this happening with a demo screen i have.

https://user-images.githubusercontent.com/5117473/140588648-938b1a60-876e-4d99-b0ff-55400eda0319.mp4

**Code sample**
I can provide this next week if needed. It would take some time to replicate what I've done and extract to a sample project. But let me know if it's not easily reproducible with more tests.

**Environment (please complete the following information):**
 - OS: iOS, Android
 - react-native-toast-message version: v2.0.1
 - react-native version: v0.64.2

**Additional context**
Likely related, but if you try to call `Toast.hide()` (like in a close button's onPress) on a toast that was shown and still on screen when a Modal becomes visible, it doesn't hide. The close button tap doesn't work. Swiping to dismiss still works, however.

This is showing a toast being shown just before the bottom modal appears, and how it can't be dismissed using `Toast.hide()` afterwards.

https://user-images.githubusercontent.com/5117473/140588724-3b9403ac-e5c5-4204-97c7-e59ac0828da8.mp4

Possibly also related, but I noticed that if I pass `autoHide: false` via `Toast.show` for the Toast instance within the Modal (which has autoHide set to true), it still auto hides despite indicating to not.

### PR DESCRIPTION
**Pull request by calintamas**
fixes #279

### CODE REVIEW

File: src/Toast.tsx [hunk 1]
```diff
@@ -28,25 +33,45 @@ const ToastRoot = React.forwardRef((props: ToastProps, ref) => {
...
-const toastRef = React.createRef<ToastRef>();
+const rootRef: ToastRefObj = {
+  current: null
+};
+const modalRef: ToastRefObj = {
+  current: null
+};
 
 export function Toast(props: ToastProps) {
+  const { setRef } = props;
```
Comment (jstheoriginal) [inline src/Toast.tsx:48]: you probably would want to do this instead, so `setRef` doesn't also get passed to `ToastRoot` when spreading in all props below.

```
export function Toast({ setRef, ...rest }: ToastProps) {
  return (
    <LoggerProvider enableLogs={false}>
      <ToastRoot ref={setRef} {...rest} />
    </LoggerProvider>
  );
}
```

Comment (calintamas) [reply src/Toast.tsx:48]: Agreed 👍

**Comment by sem-prygorodov** [comment]
@jstheoriginal Any estimate when this going to be merged and updated in npm?

**Comment by jstheoriginal** [comment]
@calintamas is the author of the PR.

**Comment by calintamas** [comment]
@sem-prygorodov It needs to suffer a few changes before it's ready for merge/publish. I'll keep you posted

**Pull request merged**